web/datapipeline/sap_systems_projector.go

```diff
@@ -0,0 +1,84 @@
+package datapipeline
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/trento/internal/sapsystem"
+	"github.com/trento-project/trento/web/entities"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+func NewSAPSystemsProjector(db *gorm.DB) *projector {
+	SAPSystemsProjector := NewProjector("sapsystems", db)
+
+	SAPSystemsProjector.AddHandler(SAPsystemDiscovery, SAPSystemsProjector_SAPSystemsDiscoveryHandler)
+
+	return SAPSystemsProjector
+}
+
+func SAPSystemsProjector_SAPSystemsDiscoveryHandler(dataCollectedEvent *DataCollectedEvent, db *gorm.DB) error {
+	decoder := getPayloadDecoder(dataCollectedEvent.Payload)
+	var discoveredSAPSystems sapsystem.SAPSystemsList
+	if err := decoder.Decode(&discoveredSAPSystems); err != nil {
+		log.Errorf("can't decode data: %s", err)
+		return err
+	}
+
+	for _, s := range discoveredSAPSystems {
+		instance := entities.SAPSystemInstance{
+			AgentID:  dataCollectedEvent.AgentID,
+			SystemID: s.Id,
+			SID:      s.SID,
+		}
+
+		if s.Type == 1 {
```

Comment:
Could we put this in a `switch/cas` fashion at least. It looks the most appropriate way